src/mon/OSDMonitor.cc

```diff
@@ -3005,6 +3010,15 @@ void OSDMonitor::get_health(list<pair<health_status_t,string> >& summary,
       summary.push_back(make_pair(HEALTH_WARN, ss.str()));
     }
 
+    if (!osds.empty()) {
+      ostringstream ss;
+      ss << "osd have removed form osdmap, but still keep in crushmap";
+      summary.push_back(make_pair(HEALTH_WARN, ss.str()));
+      if (detail) {
+        ss << " osds: [" << osds << "]";
```

Comment:
Also, you could collect detailed info here, reducing cpu and memory consumption when no details are requested.